### ISSUE TITLE
v2 cluster errors with "failed calling webhook" if created before rancher-webhook is initiated

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#32614]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Spin up rancher on the version master-360edaec8f7e236611fdd30e6798874bac267d5b-head
2. Create a node driver cluster immediately when the server is active and the local cluster is active, it immediately errors out with the following:
```
failed to create fleet-default/anupamabyo2 cluster.x-k8s.io/v1alpha4, Kind=Cluster for rke-cluster 
fleet-default/anupamabyo2: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1alpha4-cluster?timeout=10s": 
service "webhook-service" not found, failed to create fleet-default/anupamabyo2-nodepool-pool1 
cluster.x-k8s.io/v1alpha4, Kind=MachineDeployment for rke-cluster fleet-default/anupamabyo2: 
Internal error occurred: failed calling webhook "default.machinedeployment.cluster.x-k8s.io": 
Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1alpha4-machinedeployment?
timeout=10s": service "webhook-service" not found
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 
- Installation option (single install/HA): Single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): node driver/infrastructure
- Machine type (cloud/VM/metal) and specifications (CPU/memory): AWS t3.medium
- Kubernetes version (use `kubectl version`): v1.20.6+rke2r1

**Comment by anupama2501** [comment]
@Jono-SUSE-Rancher Do we have a PR for the issue?

**Comment by Jono-SUSE-Rancher** [comment]
@anupama2501 If I remember correctly this should be part of cluster provisioning and this actually might have been resolved with the work that @ibuildthecloud has been doing with his work with CP.

TLDR: It might have been fixed by virtue of some other work that was done and not necessarily tagged by a specific PR.

**Comment by anupama2501** [comment]
Verified on master-head commit d8acb21
1. Spin up the rancher server on the master-head commit. 
2. Verified the webhook is not active and created a cluster digital ocean, 1 node all roles
3. Cluster has come up active without the issue error.

**Comment by nguyenngodinh** [comment]
@anupama2501 This error still occurs. I work with version
Rancher: v2.6.0
Helm: v2.16.8-rancher1
Machine: v0.15.0-rancher69
"failed to create fleet-default/k8s-cluster-llq-ttt-177 cluster.x-k8s.io/v1alpha4, Kind=Cluster for rke-cluster fleet-default/k8s-cluster-llq-ttt-177: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1alpha4-cluster?timeout=10s": service "webhook-service" not found"

**Comment by alexhiggins732** [comment]
Same issue after trying multiple Rancher images/versions using custom and vsphere dtar Latest attempt with `rancher/rancher:v2.6-head` +`v1.21.10+rke2r1`. No issues with `k3s` or `rke` clusters.

> failed to create fleet-default/staging cluster.x-k8s.io/v1beta1, Kind=Cluster for rke-cluster fleet-default/staging: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": failed to call webhook: Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-cluster?timeout=10s": service "webhook-service" not found, failed to create fleet-default/staging-stagingpool cluster.x-k8s.io/v1beta1, Kind=MachineDeployment for rke-cluster fleet-default/staging: Internal error occurred: failed calling webhook "default.machinedeployment.cluster.x-k8s.io": failed to call webhook: Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-machinedeployment?timeout=10s": service "webhook-service" not found

**Comment by billhong-just** [comment]
Me too. 

Rancher: **v2.7.2**
```
docker run -d --restart=unless-stopped `
  -p 8080:80 -p 8443:443 `
  -v /opt/rancher:/var/lib/rancher `
  --privileged `
  rancher/rancher:v2.7.2
```


`Failed to create fleet-default/test cluster.x-k8s.io/v1beta1, Kind=Cluster for rke-cluster fleet-default/test: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": failed to call webhook: Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-cluster?timeout=10s": service "webhook-service" not found`

![image](https://user-images.githubusercontent.com/77958736/233214984-6bbf2936-7975-4ebc-befc-a869dbaf5a61.png)
![image](https://user-images.githubusercontent.com/77958736/233215033-bd1bf233-4606-4099-8210-a24d96681cad.png)

**Comment by waterliuhm** [comment]
我也遇到同样的问题，希望能尽快解决

**Comment by PeeBee66** [comment]
> Me too.
> 
> Rancher: **v2.7.2** (as a docker container)
> 
> ```
> docker run -d --restart=unless-stopped `
>   -p 8080:80 -p 8443:443 `
>   -v /opt/rancher:/var/lib/rancher `
>   --privileged `
>   rancher/rancher:v2.7.2
> ```
> 
> `Failed to create fleet-default/test cluster.x-k8s.io/v1beta1, Kind=Cluster for rke-cluster fleet-default/test: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": failed to call webhook: Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-cluster?timeout=10s": service "webhook-service" not found`
> 
> ![image](https://user-images.githubusercontent.com/77958736/233214984-6bbf2936-7975-4ebc-befc-a869dbaf5a61.png) ![image](https://user-images.githubusercontent.com/77958736/233215033-bd1bf233-4606-4099-8210-a24d96681cad.png)

You running this on Red hat?

**Comment by PeeBee66** [comment]
I am also getting the same on Rancher 2.8.2:
OS: Redhat 8.9 - Air Gapped 

This error occurs during the initial build of the cluster create 

`[ERROR} failed to create fleet-default/testcluster cluster.x-k8s.io/v1beta1, Kind=cluster for rke-cluster fleet-defualt/testcluster: internal error occured: failed calling weekhook "default.cluster.cluster.x-k8s.io": failed to call webhook: Post "https://capi-webhook-service.cattle-provisioning-capi-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-cluster?timeout=10s": service :capi-webhook-service: not found`

**Comment by hereTac** [comment]
What is the root cause this? Anyone give some clues. 2.8.2 2.8.3 no matter the versions.

**Comment by grinono** [comment]
i'm getting the same error. First though it where the certs but they look good. What else can we check?

**Comment by hereTac** [comment]
> What is the root cause this? Anyone give some clues. 2.8.2 2.8.3 no matter the versions. May be the OS or Linux kernel, any other stuff?

In my case, start the rancher server with docker, just be patient wait for the local K3S started.

**Comment by NoVeTe36** [comment]
Still meet this in rancher v2.9.1 (stable). Have to wait a long time

**Comment by HansAarneLiblik** [comment]
Trying with v2.10.1 and v2.9.2, still the same issue. Waited for 45 minutes and still nothing. I'm on air-gapped installation
